backend/test_observer/controllers/test_executions/test_executions.py

```diff
@@ -130,12 +138,10 @@ def end_test_execution(request: EndTestExecutionRequest, db: Session = Depends(g
     if test_execution is None:
         raise HTTPException(status_code=404, detail="Related TestExecution not found")
 
-    failed = any(r.status == C3TestResultStatus.FAIL for r in request.test_results)
-
-    if failed:
-        test_execution.status = TestExecutionStatus.FAILED
-    else:
-        test_execution.status = TestExecutionStatus.PASSED
+    test_cases, test_results = parse_c3_test_results(request.test_results)
+    db.add_all(test_cases)
```

Comment:
I noticed the unique constraint on `name`, but was wondering what would be the behaviour of this command be in case some of the test cases indeed exist in the database, does it complain or it only adds the new ones.

Reply:
Yup you were indeed right!